kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -242,6 +242,10 @@ static PrimaryToSecondaryMapper<HasMetadata> proxyToClusterMapper(EventSourceCon
     @VisibleForTesting
     static SecondaryToPrimaryMapper<VirtualKafkaCluster> clusterToProxyMapper(EventSourceContext<KafkaProxy> context) {
         return cluster -> {
+            // we do not want to trigger reconciliation of any proxy if the cluster has not been reconciled
+            if (!ResourcesUtil.isReconciled(cluster)) {
+                return Set.of();
```

Comment:
I think its worth logging that we are skipping the update because we are waiting for the leaf reconciler